wgsl/index.bs

```diff
@@ -992,6 +1008,26 @@ array_decoration
   </xmp>
 </div>
 
+<div class='example' heading='Access qualifier'>
+  <xmp>
+    # Storage buffers
+    var<storage> buf1 : [[access(read)]] Buffer;       # Can read, cannot write.
+    var<storage> buf2 : [[access(read_write)]] Buffer; # Can both read and write
+
+    # Access qualifier attached through a type alias.
+    type ROBuffer1 = [[access(read)]] Buffer;
+    var<storage> roBuf1 : ROBuffer1;  # Can read, cannot write
+
+    # Access qualifier directly attached to the struct definition.
```

Comment:
Why isn't this the *only* way of attaching the decoration? Why do we need two ways (on the struct declaration, or on its use in a type)

Reply:
As the example above, if you want to attach the same struct with different access controls you can't do it if you attach the access directly to the struct.

Reply:
If we care about this use case, we should just require the decorator on the usages of the type. If we don't care, we should require the decorator on the struct declaration. But why would we want to support both is not clear to me.

Reply:
Only having it on the struct would be a lot easier as then you don't have to deal with decorations on the identifier of a type declaration. We already have a struct decoration for block so this is just an extension of that.

But, then you have to declare 2 different structs if you want different access controls.

Reply:
Right. We leave the right to allow this later down the road, but for now this doesn't appear to be a use case that should be complicating the grammar and the spec.

Reply:
Just to make sure I'm understanding, you could do:

```
[[access(read)]] struct Foo {
  a : f32;
};
type B = Foo;
var c : Foo;
var d : B;
```

So, you can use the access through the type alias, but you can't set it on the type alias and you can't set it on the declaration.

This would be the only way to set the access control on a struct.

Reply:
This becomes obvious if we move the access qualifier from a decoration into the type, e.g. `storage_buffer<read> Foo {.. }`

Reply:
Moved the access qualifier to always be on the type_decl and removed from the struct.